scipy/stats/_qmc.py

```diff
@@ -83,23 +84,59 @@ def scale(sample, l_bounds, u_bounds, reverse=False):
 
     Examples
     --------
+    Transform 3 samples in the unit hypercube to bounds:
+
     >>> from scipy.stats import qmc
     >>> l_bounds = [-2, 0]
     >>> u_bounds = [6, 5]
-    >>> sample = [[0.5 , 0.5 ],
+    >>> sample = [[0.5 , 0.75],
+    ...           [0.5 , 0.5],
     ...           [0.75, 0.25]]
-    >>> qmc.scale(sample, l_bounds, u_bounds)
-    array([[2.  , 2.5 ],
+    >>> sample_scaled = qmc.scale(sample, l_bounds, u_bounds)
+    >>> sample_scaled
+    array([[2.  , 3.75],
+           [2.  , 2.5 ],
            [4.  , 1.25]])
 
+    And convert back to the unit hypercube:
+
+    >>> sample_ = qmc.scale(sample_scaled, l_bounds, u_bounds, reverse=True)
+    >>> sample_
+    array([[0.5 , 0.75],
+           [0.5 , 0.5 ],
+           [0.75, 0.25]])
+
     """
     sample = np.asarray(sample)
-    min_ = np.asarray(l_bounds)
-    max_ = np.asarray(u_bounds)
+    lower = np.atleast_1d(l_bounds)
+    upper = np.atleast_1d(u_bounds)
+
+    # Checking bounds and sample
+    if not sample.ndim == 2:
+        raise ValueError('Sample is not a 2D array')
+
+    dim = len(lower)
+    if dim != len(upper):
+        raise ValueError('Bounds do not have the same dimensions')
+
```

Comment:
You can keep this in or remove it. This may be unnecessarily strict, as it is reasonable to have e.g. the same lower bound for all dimensions (expressed as a float, 0) but different upper bounds for each dimension (expressed as an array). I think  broadcasting will do the right thing in any case.

Reply:
Explicit is better than implicit, so I would leave this. Because yes you can have someone giving `1, [2, 5]`, but we would still need a check for `[1, 5], [2, 5, 6]`.

Reply:
If you want, you can test for that using `np.broadcast_arrays`. It will produce a reasonable error message if the inputs are not broadcastable, and if they are it returns outputs that are broadcasted versions of the original arrays.

```
>>> x = [1, 5]; y = [2, 5, 6]
>>> a, b = np.broadcast_arrays(x, y)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<__array_function__ internals>", line 6, in broadcast_arrays
  File "C:\Users\matth\Anaconda3\envs\scipydev\lib\site-packages\numpy\lib\stride_tricks.py", line 258, in broadcast_arrays
    shape = _broadcast_shape(*args)
  File "C:\Users\matth\Anaconda3\envs\scipydev\lib\site-packages\numpy\lib\stride_tricks.py", line 189, in _broadcast_shape
    b = np.broadcast(*args[:32])
ValueError: shape mismatch: objects cannot be broadcast to a single shape
>>> a, b = np.broadcast_arrays(10, y)
>>> a
array([10, 10, 10])
>>> b
array([2, 5, 6])
```

Reply:
Ok I added the broadcast option and changed the tests accordingly.